### PR TITLE
[BREAKING] Clone resources when writing in and reading from the Memory

### PR DESCRIPTION
**Pull request by flovogt**
Clone resources when writing them in the memory.

JIRA: CPOUI5FOUNDATION-567

Follow-Up of https://github.com/SAP/ui5-fs/pull/259.

### CODE REVIEW

**Comment by RandomByte** [comment]
LGTM

File: test/lib/adapters/Memory_write.js [hunk 1]
```diff
@@ -25,11 +24,9 @@ test("glob resources from application.a w/o virtual base path prefix", async (t)
...
 		path: "/app/index.html"
 	});
-	await dest.write(res)
-		.then(() => dest.byGlob("/**/*.html"))
-		.then((resources) => {
-			t.is(resources.length, 1, "Found exactly one resource");
-		});
+	await dest.write(res);
+	const resources = await dest.byGlob("/**/*.html");
+	t.is(resources.length, 1, "Found exactly one resource");
```
Comment (flovogt) [inline test/lib/adapters/Memory_write.js:29]: call setContent to the written resource and check that original resource is not changed

**Comment by flovogt** [comment]
Enhance integration test "test/lib/resources.js" to call setContent to the written resource and check that original resource is not changed for both existing adapters

**Comment by flovogt** [comment]
Cherry-pick "read" PR in this one here and close the other one

File: test/lib/adapters/Memory_write.js (hunk 1 above)
Comment (flovogt) [reply test/lib/adapters/Memory_write.js:29]: done

**Review by RandomByte — CHANGES_REQUESTED** [review]

File: test/lib/adapters/Memory_write.js [hunk 2]
```diff
@@ -133,3 +131,29 @@ test("Migration of resource is executed", async (t) => {
...
+	const resource1 = await writer.byPath("/test.js");
+
+	t.is(await resource.getString(), "MyNewContent");
+	t.is(await resource1.getString(), "MyInitialContent");
+
+	await writer.write(resource);
+
+	const resource2 = await writer.byPath("/test.js");
+	t.is(await resource.getString(), "MyNewContent");
+	t.is(await resource2.getString(), "MyNewContent");
```
[the remark this answers is not in the document]
Comment (flovogt) [reply test/lib/adapters/Memory_write.js:159]: Done

**Review by RandomByte — APPROVED** [review]

**Pull request merged**